lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -1071,20 +1083,13 @@ int ReadDxcOpts(const OptTable *optionTable, unsigned flagsToInclude,
   opts.SpirvOptions.entrypointName =
       Args.getLastArgValue(OPT_fspv_entrypoint_name_EQ);
 
-  // Options not currently implemented in the SPIR-V backend.
-  // The options checked here are non-exhaustive. A thorough audit of available
-  // options and their current compatibility is needed to generate a complete
-  // list.
-  if (Args.hasFlag(OPT_spirv, OPT_INVALID, false)) {
-    if (!Args.getLastArgValue(OPT_Fd).empty()) {
-      errors << "-Fd is not currently supported with -spirv";
-      return 1;
-    }
-    if (!Args.getLastArgValue(OPT_Fre).empty()) {
-      errors << "-Fre is not currently supported with -spirv";
-      return 1;
-    }
-  }
+  // Check for use of options not implemented in the SPIR-V backend.
+  // Note: The options checked here are non-exhaustive. A thorough audit of
+  // available options and their current compatibility is needed to generate a
+  // complete list.
+  if (checkUnsupportedSpirvOption(Args, OPT_Fd, "Fd", errors) ||
+      checkUnsupportedSpirvOption(Args, OPT_Fre, "Fre", errors))
```

Comment:
What do you think about passing a list of options to check, or just moving the list of unsupported options to the helper?

Reply:
I think this is tidier now, but let me know what you think!

Reply:
Also added one more unsupported arg that's just a bool flag to make sure it functions with both opt types.